FlowCrypt/Functionality/Mail Provider/Backup Provider/Imap+Backup.swift

```diff
@@ -56,42 +60,38 @@ extension Imap: BackupProvider {
             return Data()
         }
 
-        let attachmentContext = messageContexts.flatMap { msgContext -> [AttachmentContext] in
+        let attachmentContexts = messageContexts.flatMap { msgContext -> [AttachmentContext] in
             guard let parts = msgContext.msg.attachments() as? [MCOIMAPPart] else { assertionFailure(); return [] }
             return parts.map { part in AttachmentContext(path: msgContext.path, msg: msgContext.msg, part: part) }
         }
 
         // in case there are no attachments return empty data
-        guard attachmentContext.isNotEmpty else {
+        guard attachmentContexts.isNotEmpty else {
             return Data()
         }
 
-        let dataArr = try attachmentContext.map { attContext -> Data in
-            try awaitPromise(fetchMsgAttachment(
-                in: attContext.path,
-                msgUid: attContext.msg.uid,
-                part: attContext.part
+        var dataArr: [Data] = []
+        for attachmentContext in attachmentContexts {
+            let data = try await fetchMsgAttachment(
+                in: attachmentContext.path,
+                msgUid: attachmentContext.msg.uid,
+                part: attachmentContext.part
             )
-            ) + [10] // newline
+            dataArr.append(data + [10]) // newline
         }
-
         return dataArr.joined
     }
 
-    private func fetchMsgAttachment(in folder: String, msgUid: UInt32, part: MCOIMAPPart) -> Promise<Data> {
-        Promise<Data> { [weak self] resolve, reject in
-            guard let self = self else { return reject(AppErr.nilSelf) }
-            self.imapSess?
-                .fetchMessageAttachmentOperation(
-                    withFolder: folder,
-                    uid: msgUid,
-                    partID: part.partID,
-                    encoding: part.encoding
-                )
-                .start(self.finalize("fetchMsgAtt", resolve, reject, retry: {
-                    self.fetchMsgAttachment(in: folder, msgUid: msgUid, part: part)
-                }))
-        }
+    // todo - this is likely a duplicate of another method on Imap class
+    private func fetchMsgAttachment(in folder: String, msgUid: UInt32, part: MCOIMAPPart) async throws -> Data {
+        try await execute("fetchMsgAttachment", { sess, respond in
+            sess.fetchMessageAttachmentOperation(
```

Comment:
Todo in another issue - see if we have two methods that use `sess.fetchMessageAttachmentOperation`